src/main/java/org/schemaspy/mutation/Mutation.java

```diff
@@ -0,0 +1,19 @@
+package org.schemaspy.mutation;
+
+import org.schemaspy.model.Table;
+
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * Abstraction for mutators that will modify the model.
+ * Model since a table has access to the whole object graph.
+ */
+public interface Mutation<T> {
```

Comment:
Redesign. Even though its name indicates otherwise, this interface is still a description of what the object does instead of what it is. (If the intent is to model a [mutation](https://en.wikipedia.org/wiki/Mutation), `mutate()` doesn't make sense.)

What are you trying to achieve that wouldn't be covered by a different implementation of an existing domain object? For example, a database with some extra information/different behaviour is probably still a database.

Reply:
Mutator, but I'm actually wondering if this is something we want.

Database isn't immutable, would feel better if it was.
All the Mutators/Augmentors/Enricher if we can call them that. Needs to much knowledge of the internals, so the Database interface/object becomes to exposed.

Well the database is a mess as it is, and simply cloning will just retain the mess.

But something like this.
```
Databsae database = 
  new DbWithImpliedConstraints(
    new DbWithRailsConstraints(
      new DbWithXmlAugmentation(
        new DatabaseFrom(
          new JDBCSource(connectionProperties)
         )
      )
   )
)
```

And maybe something similar to
```
new SchemaAnalyzer(
  new DbWithImpliedConstraints(
    new DbWithRailsConstraints(
      new DbWithXmlAugmentation(
        new DatabaseFrom(
          new JDBCSource(connectionProperties)
         )
      )
   ),
   new XmlMedium(),
   new JsonMedium(),
   new HtmlMedium()
).analyze();
```

So might be better to roll all of this back, and focus only on moving it out of MustacheSummaryDiagramFactory.

And figure out how to decorate the Database instead.